programs/odbc-bridge/ColumnInfoHandler.cpp

```diff
@@ -146,6 +146,10 @@ void ODBCColumnsInfoHandler::handleRequest(HTTPServerRequest & request, HTTPServ
             {
                 catalog_name = tables.table_catalog();
                 LOG_TRACE(log, "Will fetch info for table '{}.{}'", catalog_name, table_name);
+                while (tables.next())
```

Comment:
Is it more correct to do like this:
```
if (tables.next())
{
        catalog_name = tables.table_catalog();
        LOG_TRACE(log, "Will fetch info for table '{}.{}'", catalog_name, table_name);
        /// instead of "while (tables.next())"
        if (tables.next())
              throw Exception(ErrorCodes::LOGICAL_ERROR, "More than one table matched table name {} with schema {}", table_name, schema_name);
```
?
Seems like there should not be more than one matched table

Reply:
I tried it, should work, mssql returns `false` from `tables.next()`, so no unexpected results. Seems it just expects it to be checked. The only thing is that `LOGICAL_ERROR` imho, since if we got multiple results it would be not a bug in out code, but rather some unexpected behaviour in driver or some hacky user query or server setup.